internal/btf/btf.go

```diff
@@ -442,47 +442,19 @@ func (s *Spec) Program(name string, length uint64) (*Program, error) {
 // Map finds the BTF for a map.
 //
 // Returns an error if there is no BTF for the given name.
-func (s *Spec) Map(name string) (*Map, []Member, error) {
+func (s *Spec) Map(name string) (*Map, error) {
```

Comment:
Maybe just return `(*Struct, error)` here? Another option would be to get rid of this function and move it to `loadBTFMaps`. Stuffing the extra values into Map just to retrieve them later seems indeed a bit weird.

Reply:
Agreed, will give this some thought. This series of functions `(*Spec).{Map,Datasec,Program}()` could maybe use renaming to more accurately describe what they do. (e.g. `FindMap` or `FindMapByName`) That's probably best tackled in another PR.

Reply:
@lmb I've changed this to `(*Spec) Map()` to return a `Struct` and moved the Var lookup into a new `(*Spec) VarByName()`.

FWIW, this is why I would argue that moving map construction completely into `internal/btf` could make sense. `Map()` now returns a `*Struct`, but i.e. `Program()` returns a 'finished' `*Program`, which is a bit incoherent. Ideally, `Map()` would return a fully-processed `*Map` with inner maps or `ProgramSpec` pointers in the PROG_ARRAY case.

Currently, we're leaking `internal` types into `ebpf`, which is not ideal. Easier said than fixed, but food for thought. :+1: